p2p/src/lib.rs

```diff
@@ -158,6 +163,30 @@ where
     <T as NetworkingService>::PeerId: FromStr,
     <<T as NetworkingService>::PeerId as FromStr>::Err: Debug,
 {
-    let p2p = P2p::<T>::new(chain_config, p2p_config, chainstate_handle, mempool_handle).await?;
+    let p2p = P2p::<T>::new(
+        transport,
+        chain_config,
+        p2p_config,
+        chainstate_handle,
+        mempool_handle,
+    )
+    .await?;
     Ok(Box::new(p2p))
 }
+
+pub async fn new(
```

Comment:
Sorry, but how I can do that? `P2p` is generic over `NetworkingService`. And it's not possible to construct generic `Transport` now. I can implement something like `MakeTestAddress` (renamed to `TestTransport`), but it seems to me this would defeat the purpose of this PR.

Reply:
Yes, it works after making `make_p2p` a regular function (it was generic too).